app/src/scripts/dataset/tools/json/jsoneditor.store.js

```diff
@@ -118,8 +119,60 @@ let JsonEditorStore = Reflux.createStore({
       })
     } catch (e) {
       let errorMessage = 'The data you have entered is not valid JSON.'
-      this.update({ errorDetail: [e.message], errorMessage: errorMessage })
+      let detailHighlight = this.getDetailHighlight(e.message)
+      this.update({
+        errorDetail: [e.message, detailHighlight],
+        errorMessage: errorMessage,
+      })
+    }
+  },
+
+  getDetailHighlight(errorMessage) {
+    let highlightPosition = this.getHighlightPosition(errorMessage)
+    let slicePosition = highlightPosition
+    if (this.data.data[highlightPosition] !== '\n') {
+      slicePosition = highlightPosition + 1
+    }
+    let subsection = this.data.data.substring(0, slicePosition)
+
+    // split the data into separate  lines and take the last
+    const lines = subsection.split('\n')
+
+    // get the last line
+    const lastLine = lines[lines.length - 1]
+
+    // trim off the tabs at the beginning of each line
+    const tabSplit = lastLine.split('\t')
+    const lineContent = tabSplit[tabSplit.length - 1]
+    let caratPosition = lineContent.length
```

Comment:
This is minor but you're inconsistently using `let` and `const`. `let` should be reserved for when you need to reassign a variable, which is pretty rare (only one of the `let` declarations in this function looks needed).